pkg/workload/workload.go

```diff
@@ -1104,9 +1104,28 @@ func HasQuotaReservation(w *kueue.Workload) bool {
 
 // UpdateReclaimablePods updates the ReclaimablePods list for the workload with SSA.
 func UpdateReclaimablePods(ctx context.Context, c client.Client, w *kueue.Workload, reclaimablePods []kueue.ReclaimablePod) error {
```

Comment:
Actually can we revisit to use only PatchAdmissionStatus as the only function to patch workload status?
Or we could introduce PatchStatus to deduplicate, then we could also get rid of SetConditionAndUpdate which I find redundant. All the functions look similar.

Reply:
Prototyped it here https://github.com/kubernetes-sigs/kueue/pull/8063.